src/main/java/practice/StreamPractice.java

```diff
@@ -23,7 +34,12 @@ public int findMinEvenNumber(List<String> numbers) {
      * But before that subtract 1 from each element on an odd position (having the odd index).
      */
     public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+        return IntStream.range(0, numbers.size())
+                .map(num -> num % 2 == 1
+                        ? numbers.get(num) - 1 : numbers.get(num))
+                .filter(odd -> odd % 2 == 1)
```

Comment:
```suggestion
                .filter(odd -> odd % 2 != 0)
```
better compare with 0
because -3 % 2 == -1